File: pi-repomap/ts-src/engine/git.ts
import { execa } from "execa";

export async function gitHead(repoPath: string): Promise<string> {
  try {
    const result = await execa("git", ["rev-parse", "HEAD"], {
      cwd: repoPath,
      timeout: 5_000,
    });
    return result.stdout.trim();
  } catch {
    return "nogit";
  }
}

File: pi-repomap/src/scope/engine/git.py
from __future__ import annotations

import subprocess
from typing import List


def git_head(repo_path: str) -> str:
    try:
        result = subprocess.run(
            ["git", "rev-parse", "HEAD"], cwd=repo_path,
            capture_output=True, text=True, timeout=5,
        )
        if result.returncode == 0:
            return result.stdout.strip()
    except (subprocess.TimeoutExpired, FileNotFoundError, OSError):
        pass
    return "nogit"
